core/dbt/tests/tables.py

```diff
@@ -0,0 +1,339 @@
+from dbt.tests.util import run_sql
+from dbt.context import providers
+from unittest.mock import patch
+from contextlib import contextmanager
+
+
+class TableComparison:
```

Comment:
I would like to challenge this class a bit. The way I see it is that this "new way of testing" removes some of the "magic" of the "old way". The risk with using such a class is that magic is being introduced - in a different form.